tests/integration/mongodb/test_charm.py

```diff
@@ -205,119 +207,118 @@ async def test_set_password_action(ops_test: OpsTest, substrate: Substrate) -> N
 
 
 @pytest.mark.abort_on_fail
-async def test_reset_and_get_password_secret_same_as_cli(ops_test: OpsTest) -> None:
-    """Test verifies that we can set and retrieve the correct password using Juju 3.x secrets."""
-    app_name = await get_app_name(ops_test)
-    new_password = str(uuid4())
-
-    # Resetting existing password
-    leader_id = await get_leader_id(ops_test)
-    await set_password(ops_test, unit_id=leader_id, username="monitor", password=new_password)
-
-    # Getting back the pw programmatically
-    password = await get_password(ops_test, username="monitor")
-
-    secret_label = f"{app_name}.app"
-
-    # Getting back the pw from juju CLI
-    complete_command = f"show-secret {secret_label} --reveal --format=json"
-    _, stdout, _ = await ops_test.juju(*complete_command.split())
-    data = one(json.loads(stdout).values())
-
-    assert password == new_password
-
-    assert data["content"]["Data"]["monitor-password"] == password
-
-
 async def test_empty_password(ops_test: OpsTest) -> None:
     """Test that the password can't be set to an empty string."""
-    leader_id = await get_leader_id(ops_test)
+    app_name = await get_app_name(ops_test)
+    password1 = MONITOR_PASSWORD
+    await set_password(ops_test, username=MONITOR_USERNAME, password=" ", app_name=app_name)
 
-    password1 = await get_password(ops_test, username="monitor")
-    await set_password(ops_test, unit_id=leader_id, username="monitor", password="")
-    password2 = await get_password(ops_test, username="monitor")
+    app = ops_test.model.applications[app_name]
+    await ops_test.model.block_until(*[lambda: app.status == "blocked"], timeout=1000)
+    assert app.status_message == "Invalid secret in system-users config."
 
+    password2 = await get_password(ops_test, username=MONITOR_USERNAME, app_name=app_name)
     # The password remained unchanged
     assert password1 == password2
 
+    # Restore valid password
+    await set_password(
+        ops_test, username=MONITOR_USERNAME, password=MONITOR_PASSWORD, app_name=app_name
+    )
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=1000)
+
 
 @pytest.mark.abort_on_fail
 async def test_no_password_change_on_invalid_password(ops_test: OpsTest) -> None:
     """Test that in general, there is no change when password validation fails."""
-    leader_id = await get_leader_id(ops_test)
-    password1 = await get_password(ops_test, username="monitor")
+    app_name = await get_app_name(ops_test)
+    password1 = MONITOR_PASSWORD
+
+    # The password has to be maximum 4096-character long
+    await set_password(
+        ops_test,
+        username=MONITOR_USERNAME,
+        password="c" * 4097,
+        app_name=app_name,
+    )
+    app = ops_test.model.applications[app_name]
+    await ops_test.model.block_until(*[lambda: app.status == "blocked"], timeout=1000)
+    assert app.status_message == "Invalid secret in system-users config."
```

Comment:
If we're going to repeat this more than 3 times, create a helper in `common.py` helpers file and use it anytime you want to block until a certain status.